### PR TITLE
Fix `/services/{svc}/{group}/config` endpoint in http-gateway and add docs

### PR DESCRIPTION
**Pull request by reset**
This PR contains some changes to the http-gateway in the supervisor:

* Re-enable the `/services/{svc}/{group}/config` endpoint
* Add `/services/{svc}/{group}` and `/services/{svc}/{group}/{org}` endpoint
* Update raml to 1.0 and add a bunch of docs to it

tenor-41417425

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by reset** [comment]
Fixes #2250

File: components/sup/src/http_gateway.rs [hunk 1]
```diff
@@ -244,6 +259,32 @@ fn health(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+fn service(req: &mut Request) -> IronResult<Response> {
+    let state = req.get::<persistent::Read<ManagerFs>>().unwrap();
```
Comment (chefsalim) [inline components/sup/src/http_gateway.rs]: The bulk of code in the service and config functions is the same.. maybe add a helper function with the common code?

File: components/sup/src/http_gateway.rs [hunk 2]
```diff
@@ -244,6 +259,32 @@ fn health(req: &mut Request) -> IronResult<Response> {
...
+    let service_group = match build_service_group(req) {
+        Ok(sg) => sg,
+        Err(_) => return Ok(Response::with(status::BadRequest)),
+    };
+    match File::open(&state.services_data_path) {
```
Comment (mwrock) [inline components/sup/src/http_gateway.rs]: Maybe consider a `service_from_file` function to dry up the service deserialization but not a show stopper.

**Comment by reset** [comment]
@mwrock @chefsalim great suggestion, I went ahead and added that helper and pushed it up!

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/230127233).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/230127233).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**